noir_stdlib/src/array/mod.nr

```diff
@@ -136,6 +136,29 @@ impl<T, let N: u32> [T; N] {
         }
         ret
     }
+
+    /// Concatenates this array with another array.
+    ///
+    /// Example:
+    ///
+    /// ```noir
+    /// fn main() {
+    ///     let arr1 = [1, 2, 3, 4];
+    ///     let arr2 = [6, 7, 8, 9, 10, 11];
+    ///     let concatenated_arr = arr1.concat(arr2);
+    ///     assert(concatenated_arr == [1, 2, 3, 4, 6, 7, 8, 9, 10, 11]);
+    /// }
+    /// ```
+    pub fn concat<let M: u32>(self, array2: [T; M]) -> [T; N + M] {
+        let mut result = [self[0]; N + M];
+        for i in 1..N {
```

Comment:
```suggestion
        let mut result = [crate::mem::zeroed(); N + M];
        for i in 0..N {
```

Reply:
Thanks - why is this preferred?

Reply:
I think otherwise this doesn't work if you concat an empty array with another array (I didn't try it, but my guess is that `self[0]` would give an error)

Reply:
Ah, yes that makes sense, thanks!